lib/autoload/component_not_found_error.rb

```diff
@@ -0,0 +1,5 @@
+class ComponentNotFoundError < StandardError
+  def message
```

Comment:
this should be set in an initializer and passed to super. remember to freeze the string.